synapse/config/federation.py

```diff
@@ -36,7 +49,7 @@ def read_config(self, config, **kwargs):
             for domain in federation_domain_whitelist:
                 self.federation_domain_whitelist[domain] = True
 
-        ip_range_blacklist = config.get("ip_range_blacklist", [])
+        ip_range_blacklist = config.get("ip_range_blacklist") or DEFAULT_IP_RANGE_BLACKLIST
```

Comment:
I'd rather "empty list" didn't get treated the same as `None`/ not set. I feel like if people deliberately set an empty list, we should honour that? Or maybe error out?

Reply:
I went back and forth on this and originally implemented it that way. I ended up doing it this way so that it matches our sample config style where an empty option is the same as not having it, but now I'm realizing that really only works for maps, not lists.